src/richie-front/js/utils/filters/getFilterFromState.ts

```diff
@@ -22,21 +23,16 @@ export function getFilterFromState(
       state.resources.courses.currentQuery.facets) ||
     {};
 
-  switch (machineName) {
-    // Use the facets to build the values for resource based filters
-    case 'organizations':
-    case 'subjects':
-      return {
-        ...state.filterDefinitions[machineName],
-        values: getFacetedValues(state, facets, machineName),
-      };
-
-    // Values from state are usable as-is for hardcoded filters
-    default:
-      return {
-        ...state.filterDefinitions[machineName],
-        values: getHarcodedFilterFacetedValues(facets, machineName),
-      };
+  if (isResourceBasedFilterGroupName(machineName)) {
```

Comment:
I know I'm the one who suggested an `if/else` but here a ternary would probably be more expressive and clearer